packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,327 @@
+    };
+
+    private setupTrackConstraints = (): void => {
+        this.trackWidth = this.track.clientWidth;
+        this.trackMinWidth = this.track.getBoundingClientRect().left;
+    };
+
+    private setupListeners = (): void => {
+        this.addEventListener("keydown", this.keypressHandler);
+        this.addEventListener("mousedown", this.clickHandler);
+        this.thumb.addEventListener("mousedown", this.handleThumbMouseDown);
+    };
+
+    private setupClassForOrientation = (): void => {
+        if (this.orientation === SliderOrientation.horizontal) {
+            this.root.classList.add("slider-horizontal");
+        } else {
+            this.root.classList.add("slider-vertical");
+        }
+    };
+
+    private setupDefaultValue = (): void => {
+        if (this.value === "") {
+            this.value = `${this.convertToConstrainedValue((this.max - this.min) / 2)}`;
+            this.updateForm();
+        }
+    };
+
+    private updateForm = (): void => {
+        this.proxy.value = this.value;
+    };
+
+    /**
+     *  Handle mouse moves during a thumb drag operation
+     */
+    private handleThumbMouseDown = (event: MouseEvent): void => {
+        if (this.readOnly || this.disabled || event.defaultPrevented) {
+            return;
+        }
+        event.preventDefault();
+        (event.target as HTMLElement).focus();
+        window.addEventListener("mouseup", this.handleWindowMouseUp);
+        window.addEventListener("mousemove", this.handleMouseMove);
+        this.isDragging = true;
+    };
+
+    /**
+     *  Handle mouse moves during a thumb drag operation
+     */
+    private handleMouseMove = (e: MouseEvent): void => {
+        if (this.readOnly || this.disabled || e.defaultPrevented) {
+            return;
+        }
+        // update the value based on current position
+        this.value = `${this.calculateNewValue(e.pageX)}`;
+        this.updateForm();
+    };
+
+    private calculateNewValue = (rawValue: number): number => {
+        // update the value based on current position
+        const newPosition = convertPixelToPercent(
+            rawValue,
+            this.trackMinWidth,
+            this.trackWidth,
+            this.direction
+        );
+        const newValue: number = (this.max - this.min) * newPosition + this.min;
+        return this.convertToConstrainedValue(newValue);
+    };
+
+    /**
+     * Handle a window mouse up during a drag operation
+     */
+    private handleWindowMouseUp = (event: MouseEvent): void => {
+        this.stopDragging();
+    };
+
+    private stopDragging = (): void => {
+        this.isDragging = false;
+        window.removeEventListener("mouseup", this.handleWindowMouseUp);
+        window.removeEventListener("mousemove", this.handleMouseMove);
+    };
+
+    private clickHandler = (e: MouseEvent) => {
+        if (!this.disabled && !this.readOnly) {
+            let trackElement: any = this.shadowRoot!.querySelector(".track");
```

Comment:
Question: could you use a ref for this?

Reply:
yes, I think I did this before I started using refs. will change